### PR TITLE
Adding capability to instantiate a vApp with custom disk size

### PR DESCRIPTION
**Pull request by rocknes**
Added capability to instantiate a vApp with custom disk size

vApp instantiation can now happen for VDCs with no network,

Refactored the function vapp_instantiate() to rename few variables and moved around sections of code to tightly follow the order of creation of various elements of vAppTemplateInstantiationParam.

ToDo : Unit tests
Manually tested that the functionality is working as per design.

### CODE REVIEW

File: pyvcloud/vcd/vdc.py [hunk 1]
```diff
@@ -69,6 +68,7 @@ def delete_vapp(self, name, force=False):
         href = self.get_resource_href(name)
         return self.client.delete_resource(href, force)
 
+    # refer to http://pubs.vmware.com/vcd-820/index.jsp?topic=%2Fcom.vmware.vcloud.api.sp.doc_27_0%2FGUID-BF9B790D-512E-4EA1-99E8-6826D4B8E6DC.html
```
Comment (pacogomez) [inline pyvcloud/vcd/vdc.py]: insert `NOQA` at the beginning of the comment to avoid flake8 errors

File: pyvcloud/vcd/vdc.py [hunk 2]
```diff
@@ -94,158 +96,190 @@ def instantiate_vapp(self,
...
+            for item in items:
+                if ((memory is not None) and (memory_params is None)):
+                    if item['{' + NSMAP['rasd'] + '}ResourceType'] == 4:  # NOQA
+                        item['{' + NSMAP['rasd'] + '}ElementName'] = '%s MB of memory' % memory  # NOQA
+                        item['{' + NSMAP['rasd'] + '}VirtualQuantity'] = memory  # NOQA
+                        memory_params = item
+                        virtual_hardwire_section.append(memory_params)
+
+                if ((cpu is not None) and (cpu_params is None)):
+                    if item['{' + NASMAP['rasd'] + '}ResourceType'] == 3:  # NOQA
```
Comment (pacogomez) [inline pyvcloud/vcd/vdc.py]: typo, it should say `NSMAP`

Comment (rocknes) [reply pyvcloud/vcd/vdc.py]: Fixed typo

File: tests/vcd_vapp.py [hunk 3]
```diff
@@ -116,5 +116,56 @@ def test_110_instantiate_vapp_identical(self):
...
+        vdc = VDC(self.client, href=v.get('href'))
+        assert self.config['vcd']['vdc'] == vdc.get_resource().get('name')
+        result = vdc.instantiate_vapp(self.config['vcd']['vapp'],
+                                      self.config['vcd']['catalog'],
+                                      self.config['vcd']['template'],
+                                      network=self.config['vcd']['network'],
+                                      fence_mode='bridged',
+                                      deploy=True,
+                                      power_on=False,
+                                      disk_size=self.config['vcd']['disk_size_new'])
```
Comment (anusuyar) [inline tests/vcd_vapp.py]: Add these new config params to config.yml so that others can run these tests just by changing the param value.

Comment (anusuyar) [inline tests/vcd_vapp.py]: Add new config params(disk_size_new here) in config.yml. Others can run these tests after just changing the value.

**Pull request merged**